### PR TITLE
Use CultureInvariant string conversion for Guids

### PR DESCRIPTION
**Pull request by Bond-009**
What it says on the tin.

### CODE REVIEW

**Comment by cvium** [comment]
Why?

**Comment by Bond-009** [comment]
@cvium consistency across different systems, just good practice.

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs [hunk 1]
```diff
@@ -76,7 +72,6 @@ public Task RunAsync()
             _sessionManager.PlaybackStart += _sessionManager_PlaybackStart;
             _sessionManager.PlaybackStopped += _sessionManager_PlaybackStopped;
 
-            //_subManager.SubtitlesDownloaded += _subManager_SubtitlesDownloaded;
```
Comment (cvium) [inline Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: I think we should re-enable it instead

Comment (Bond-009) [reply Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: The event got removed as nothing ever triggers it.

Comment (LogicalPhallacy) [reply Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: Was it there for plugins to use by any chance?

File: Emby.Server.Implementations/Dto/DtoService.cs [hunk 2]
```diff
@@ -89,11 +90,10 @@ public BaseItemDto[] GetBaseItemDtos(IEnumerable<BaseItem> items, int itemCount,
             var channelTuples = new List<Tuple<BaseItemDto, LiveTvChannel>>();
 
             var index = 0;
-            var allCollectionFolders = _libraryManager.GetUserRootFolder().Children.OfType<Folder>().ToList();
```
Comment (LogicalPhallacy) [inline Emby.Server.Implementations/Dto/DtoService.cs]: Whats the deal with removing this, I'm not necessarily opposed but it seems odd in a PR about CultureInvariant

File: MediaBrowser.Providers/Manager/ProviderManager.cs [hunk 3]
```diff
@@ -977,7 +977,7 @@ public void OnRefreshComplete(BaseItem item)
 
         public void OnRefreshProgress(BaseItem item, double progress)
         {
-            //_logger.LogInformation("OnRefreshProgress {0} {1}", item.Id.ToString("N"), progress);
+            //_logger.LogInformation("OnRefreshProgress {0} {1}", item.Id.ToString("N", CultureInfo.InvariantCulture), progress);
```
Comment (LogicalPhallacy) [inline MediaBrowser.Providers/Manager/ProviderManager.cs]: This commented line should probably either be put behind a debug compiler directive or deleted.

File: jellyfin.ruleset [hunk 4]
```diff
@@ -15,6 +15,13 @@
     <Rule Id="SA1633" Action="None" />
   </Rules>
   <Rules AnalyzerId="Microsoft.CodeAnalysis.FxCopAnalyzers" RuleNamespace="Microsoft.Design">
+    <!-- disable warning CA1812: internal class that is apparently never instantiated.
+        If so, remove the code from the assembly.
+        If this class is intended to contain only static members, make it static -->
+    <Rule Id="CA1812" Action="Info" />
+    <!-- disable warning CA1822: Member does not access instance data and can be marked as static -->
```
Comment (LogicalPhallacy) [inline jellyfin.ruleset]: I get removing CA1812, but CA1822 might be worth keeping as a warning.

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs [hunk 5]
```diff
@@ -31,27 +31,23 @@ public class ActivityLogEntryPoint : IServerEntryPoint
...
-
-        private readonly ILibraryManager _libraryManager;
         private readonly ISubtitleManager _subManager;
         private readonly IUserManager _userManager;
         private readonly IServerConfigurationManager _config;
         private readonly IServerApplicationHost _appHost;
         private readonly IDeviceManager _deviceManager;
 
-        public ActivityLogEntryPoint(ISessionManager sessionManager, IDeviceManager deviceManager, ITaskManager taskManager, IActivityManager activityManager, ILocalizationManager localization, IInstallationManager installationManager, ILibraryManager libraryManager, ISubtitleManager subManager, IUserManager userManager, IServerConfigurationManager config, IServerApplicationHost appHost)
+        public ActivityLogEntryPoint(ISessionManager sessionManager, IDeviceManager deviceManager, ITaskManager taskManager, IActivityManager activityManager, ILocalizationManager localization, IInstallationManager installationManager, ISubtitleManager subManager, IUserManager userManager, IServerConfigurationManager config, IServerApplicationHost appHost)
```
Comment (JustAMan) [inline Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: please don't change unrelated things in single PR, let's stick to Guids here

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs (hunk 1 above)
Comment (JustAMan) [reply Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: And yes, I agree here that keeping it might be worth it for activity DB and some reports, maybe plugins, too.

File: Emby.Server.Implementations/Dto/DtoService.cs (hunk 2 above)
Comment (JustAMan) [reply Emby.Server.Implementations/Dto/DtoService.cs]: I second this, let's not mix different things in one PR.

File: Emby.Server.Implementations/Security/AuthenticationRepository.cs [hunk 6]
```diff
@@ -15,13 +15,9 @@ namespace Emby.Server.Implementations.Security
 {
     public class AuthenticationRepository : BaseSqliteRepository, IAuthenticationRepository
     {
-        private readonly IServerConfigurationManager _config;
```
Comment (JustAMan) [inline Emby.Server.Implementations/Security/AuthenticationRepository.cs]: why remove in an unrelated PR? pls don't

File: Emby.Server.Implementations/Services/StringMapTypeDeserializer.cs [hunk 7]
```diff
@@ -110,9 +110,9 @@ public static string LeftPart(string strVal, char needle)
         }
     }
 
-    internal class TypeAccessor
+    internal static class TypeAccessor
```
Comment (JustAMan) [inline Emby.Server.Implementations/Services/StringMapTypeDeserializer.cs]: pls don't change stuff that is unrelated to main PR. at the very least edit the description... but I would really prefer separate PRs.

File: MediaBrowser.Controller/Entities/UserView.cs [hunk 8]
```diff
@@ -2,7 +2,6 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
-using MediaBrowser.Controller.Playlists;
```
Comment (JustAMan) [inline MediaBrowser.Controller/Entities/UserView.cs]: again unrelated change, why?

File: MediaBrowser.Controller/Entities/UserViewBuilder.cs [hunk 9]
```diff
@@ -21,17 +22,15 @@ public class UserViewBuilder
         private readonly IUserDataManager _userDataManager;
         private readonly ITVSeriesManager _tvSeriesManager;
         private readonly IServerConfigurationManager _config;
-        private readonly IPlaylistManager _playlistManager;
```
Comment (JustAMan) [inline MediaBrowser.Controller/Entities/UserViewBuilder.cs]: again unrelated change

File: Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Activity/ActivityLogEntryPoint.cs]: The event doesn't even exist anymore, we cant keep it in.

File: jellyfin.ruleset (hunk 4 above)
Comment (Bond-009) [reply jellyfin.ruleset]: We keep them as info, we plan on failing on warnings.

**Comment by anthonylavado** [comment]
@Bond-009 Just need to fix up the conflicts now

**Comment by dkanada** [comment]
@Bond-009 are there too many conflicts at this point to continue working on this pull request?

**Comment by dkanada** [comment]
@Bond-009 are there too many conflicts or can we salvage this pull request?

File: MediaBrowser.Providers/Manager/ProviderManager.cs (hunk 3 above)
Comment (dkanada) [reply MediaBrowser.Providers/Manager/ProviderManager.cs]: I agree, the three `LogInformation` instances should get changed to `LogDebug` and enabled. The exception below should also get changed to either warn or error so we can start getting some logging for that issue.

**Comment by Bond-009** [comment]
Ref: #712

File: MediaBrowser.Providers/Manager/ProviderManager.cs (hunk 3 above)
Comment (joshuaboniface) [reply MediaBrowser.Providers/Manager/ProviderManager.cs]: 100% yes. We need more debug logs to catch issues anyways, no sense commenring out valid logging when we have a lower prio to throw it under.

Comment (Bond-009) [reply MediaBrowser.Providers/Manager/ProviderManager.cs]: Yeah, but that's out of the scope of this PR imo
